core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -50,3 +65,16 @@ private[spark] case class ExecutorDeadException(message: String)
 private[spark] class SparkUpgradeException(version: String, message: String, cause: Throwable)
   extends RuntimeException("You may get a different result due to the upgrading of Spark" +
     s" $version: $message", cause)
+
+/**
+ * Arithmetic exception thrown from Spark with an error class.
+ */
+private[spark] class SparkArithmeticException(
```

Comment:
`private[spark]`? Could we make the new standardized error classes public to 3rd-party develoeprs?

Reply:
Good idea. Done - thanks for the review!